pkg/resources/saml2_integration.go

```diff
@@ -0,0 +1,543 @@
+			}
+		case "SAML2_X509_CERT":
+			if err := d.Set("saml2_x509_cert", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_SP_INITIATED_LOGIN_PAGE_LABEL":
+			if err := d.Set("saml2_sp_initiated_login_page_label", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_ENABLE_SP_INITIATED":
+			if err := d.Set("saml2_enable_sp_initiated", helpers.StringToBool(value)); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_SNOWFLAKE_X509_CERT":
+			if err := d.Set("saml2_snowflake_x509_cert", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_SIGN_REQUEST":
+			if err := d.Set("saml2_sign_request", helpers.StringToBool(value)); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_REQUESTED_NAMEID_FORMAT":
+			if err := d.Set("saml2_requested_nameid_format", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_POST_LOGOUT_REDIRECT_URL":
+			if err := d.Set("saml2_post_logout_redirect_url", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_FORCE_AUTHN":
+			if err := d.Set("saml2_force_authn", helpers.StringToBool(value)); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_SNOWFLAKE_ISSUER_URL":
+			if err := d.Set("saml2_snowflake_issuer_url", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_SNOWFLAKE_ACS_URL":
+			if err := d.Set("saml2_snowflake_acs_url", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_SNOWFLAKE_METADATA":
+			if err := d.Set("saml2_snowflake_metadata", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_DIGEST_METHODS_USED":
+			if err := d.Set("saml2_digest_methods_used", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SAML2_SIGNATURE_METHODS_USED":
+			if err := d.Set("saml2_signature_methods_used", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "ALLOWED_USER_DOMAINS":
+			value = strings.TrimLeft(value, "[")
+			value = strings.TrimRight(value, "]")
+			elems := strings.Split(value, ",")
+			if value == "" {
+				elems = nil
+			}
+			if err := d.Set("allowed_user_domains", elems); err != nil {
+				return diag.FromErr(err)
+			}
+		case "ALLOWED_EMAIL_PATTERNS":
+			value = strings.TrimLeft(value, "[")
+			value = strings.TrimRight(value, "]")
+			elems := strings.Split(value, ",")
+			if value == "" {
+				elems = nil
+			}
+			if err := d.Set("allowed_email_patterns", elems); err != nil {
+				return diag.FromErr(err)
+			}
+		default:
+			log.Printf("[WARN] unexpected property %v returned from Snowflake", name)
+		}
+	}
+
+	return nil
+}
+
+func UpdateContextSAMLIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+	set, unset := sdk.NewSaml2IntegrationSetRequest(), sdk.NewSaml2IntegrationUnsetRequest()
+
+	if d.HasChange("enabled") {
```

Comment:
why not unset? (if this is the buggy one, we have to be explicit with it)

And this is for all others too.

Reply:
@sfc-gh-jcieslak what about this?

Reply:
Unset doesn't work so in the case of unset I set it to the Snowflake default value